### PR TITLE
Add readyToPayoutInstruction property

### PR DESCRIPTION
**Pull request by i-yasko**
## Summary
In this PR added `readyToPayoutInstruction ` to manage payout behaviour of payment gateway

## Checklist

- [x] Writing style
- [x] API design standards

### CODE REVIEW

File: openapi/components/schemas/GatewayAccount.yaml [hunk 1]
```diff
@@ -369,6 +369,26 @@ properties:
       do-nothing: |-
         Does nothing except return an approved `setup` transaction.
         This is the default behavior.
+  readyToPayoutInstruction:
+    type: string
+    default: none
+    description: |-
+      Defines a mode of gateway payout behaviour.
```
Comment (ashkarpetin) [inline openapi/components/schemas/GatewayAccount.yaml:376]: Perhaps this need clarification that it applies for StorefrontReadyToPayout and does not affect PostPayout (transaction type `credit`)?

File: openapi/components/schemas/GatewayAccount.yaml [hunk 2]
```diff
@@ -370,10 +370,13 @@ properties:
...
   readyToPayoutInstruction:
-    type: string
+    type:
+      - 'string'
+      - 'null'
     default: none
     description: |-
-      Defines a mode of gateway payout behaviour.
+      Defines a mode of gateway payout behaviour for [StorefrontReadyToPayout](https://all-rebilly.redoc.ly/tag/Purchases/operation/StorefrontPostReadyToPayout).
+      If a `null`, a gateway does not support payout operation.
```
Comment (noon-dawg) [inline openapi/components/schemas/GatewayAccount.yaml:379]: suggestion: for readability 
```suggestion
      If this value is `null`, payouts are not supported by the gateway.
```

File: openapi/components/schemas/GatewayAccount.yaml [hunk 3]
```diff
@@ -369,6 +369,29 @@ properties:
       do-nothing: |-
         Does nothing except return an approved `setup` transaction.
         This is the default behavior.
+  readyToPayoutInstruction:
+    type:
+      - 'string'
+      - 'null'
+    default: none
+    description: |-
+      Defines a mode of gateway payout behaviour for [StorefrontReadyToPayout](https://all-rebilly.redoc.ly/tag/Purchases/operation/StorefrontPostReadyToPayout).
```
Comment (noon-dawg) [inline openapi/components/schemas/GatewayAccount.yaml:378]: suggestion: for readability

I get a 404 when I try to visit the list [the URL](https://all-rebilly.redoc.ly/tag/Purchases/operation/StorefrontPostReadyToPayout)

```suggestion
      Defines the mode of gateway payout behaviour for the [Ready to payout](https://all-rebilly.redoc.ly/tag/Purchases/operation/StorefrontPostReadyToPayout) operation.
```

File: openapi/components/schemas/GatewayAccount.yaml [hunk 4]
```diff
@@ -369,6 +369,29 @@ properties:
...
+    enum:
+      - all
+      - covered-payout
+      - approved-payment
+      - none
+    x-enumDescriptions:
+      all: |-
+        Allow any amount to payout through gateway.
+      covered-payout: |-
+        Only payout amount of gateway what has been paid into gateway.
```
Comment (noon-dawg) [inline openapi/components/schemas/GatewayAccount.yaml:389]: suggestion: to use the same text as the [UI copy](https://github.com/Rebilly/rebilly/pull/2354/files#diff-ca06702ad82a26622aa233ca38f7ac936dfa2e514eb1c8315cd29f6a8c224289)
```suggestion
        Payout from this gateway if it previously processed a payment for the same, 
        or a greater, amount.
```

File: openapi/components/schemas/GatewayAccount.yaml [hunk 5]
```diff
@@ -369,6 +369,29 @@ properties:
...
+      approved-payment: |-
+        Payout any amount as long as there was a previously approved transaction for this customer and currency.
+        via this gateway
```
Comment (noon-dawg) [inline openapi/components/schemas/GatewayAccount.yaml:392]:
```suggestion
        Payout any amount from this gateway if it processed an earlier payment. 
        The customer must have a previously approved transaction, 
        in the same currency, on this gateway.
```

File: openapi/components/schemas/GatewayAccount.yaml [hunk 6]
```diff
@@ -369,6 +369,29 @@ properties:
...
+      none:
+        Gateway not allowed for payout. This will be the default value, merchant will need to opt into payouts.
```
Comment (noon-dawg) [inline openapi/components/schemas/GatewayAccount.yaml:394]:
```suggestion
        Do not allow any payouts with this gateway. 
        This is the default value, 
        merchants must opt into payouts.
```

File: openapi/components/schemas/GatewayAccount.yaml [hunk 7]
```diff
@@ -369,6 +369,29 @@ properties:
...
+      Defines a mode of gateway payout behaviour for [StorefrontReadyToPayout](https://all-rebilly.redoc.ly/tag/Purchases/operation/StorefrontPostReadyToPayout).
+      If this value is `null`, payouts are not supported by the gateway.
+    enum:
+      - all
+      - covered-payout
+      - approved-payment
+      - none
+    x-enumDescriptions:
+      all: |-
+        Allow any amount to payout through gateway.
```
Comment (noon-dawg) [inline openapi/components/schemas/GatewayAccount.yaml:387]:
```suggestion
        Payout any amount from this gateway.
```

**Review by noon-dawg — DISMISSED** [review]
Looks good on the docs side

**Review by ashkarpetin — DISMISSED** [review]

File: openapi/components/schemas/GatewayAccount.yaml [hunk 8]
```diff
@@ -369,6 +369,33 @@ properties:
       do-nothing: |-
         Does nothing except return an approved `setup` transaction.
         This is the default behavior.
+  readyToPayoutInstruction:
+    type:
+      - 'string'
+      - 'null'
```
Comment (ashkarpetin) [inline openapi/components/schemas/GatewayAccount.yaml:375]: I think it should be nullable, default value is `none`

**Review by ashkarpetin — APPROVED** [review]

**Review by justnero — APPROVED** [review]

**Pull request merged**